test/testsizeof.cpp

```diff
@@ -162,6 +163,76 @@ class TestSizeof : public TestFixture {
                       "[test.cpp:5]: (warning, inconclusive) Found calculation inside sizeof().\n", errout.str());
     }
 
+    void sizeofFunction() {
+        check("class Foo\n"
+              "{\n"
+              "    int bar() { return 1; };\n"
+              "}\n"
+              "Foo f;int a,sizeof(f.bar())");
+        ASSERT_EQUALS("[test.cpp:5]: (warning) Found function call inside sizeof().\n", errout.str());
+
+        check("class Foo\n"
+              "{\n"
+              "    int bar() { return 1; };\n"
+              "    int bar() const { return 1; };\n"
+              "}\n"
+              "Foo f;int a,sizeof(f.bar())");
+        ASSERT_EQUALS("", errout.str());
+
+        check("class Foo\n"
+              "{\n"
+              "    int bar() { return 1; };\n"
+              "}\n"
+              "Foo * fp;int a,sizeof(fp->bar())");
+        ASSERT_EQUALS("[test.cpp:5]: (warning) Found function call inside sizeof().\n", errout.str());
+
+        check("#define foo() int\n"
+              "int a,sizeof(foo())");
+        ASSERT_EQUALS("", errout.str());
```

Comment:
this is not technically correct. The testcases are not preprocessed. so I believe that during testing the checker would see `# define foo ( ) int ...`. But when you actually run it on this code, it will not see the define at all.

Imho you can just remove the first line and assert that the checker will not warn when a `sizeof(foo())` is seen and it does not know what foo is.